src/main/webapp/app/exercise/result/result.utils.ts

```diff
@@ -369,39 +369,33 @@ export const getResultIconClass = (result: Result | undefined, templateStatus: R
 
 /**
  * Returns true if the specified result is preliminary.
- * @param result the result. It must include a participation and exercise.
+ * @param result the result.
+ * @param participation the participation
  */
-export const resultIsPreliminary = (result: Result) => {
-    const exerciseType = result.submission?.participation?.exercise?.type;
+export const resultIsPreliminary = (result: Result, participation: Participation) => {
+    const exerciseType = participation?.exercise?.type;
     if (exerciseType === ExerciseType.TEXT || exerciseType === ExerciseType.MODELING) {
         return result.assessmentType === AssessmentType.AUTOMATIC_ATHENA;
-    } else
-        return (
-            result.submission?.participation &&
-            isProgrammingExerciseStudentParticipation(result.submission.participation) &&
-            isResultPreliminary(result, result.submission?.participation?.exercise as ProgrammingExercise)
-        );
+    } else return isProgrammingExerciseStudentParticipation(participation) && isResultPreliminary(result, participation, participation?.exercise as ProgrammingExercise);
 };
 
 /**
  * Returns true if the specified result is a student Participation
- * @param result the result.
+ * @param participation the participation
  */
-export const isStudentParticipation = (result: Result) => {
-    return Boolean(
-        result.submission?.participation &&
-            result.submission.participation.type !== ParticipationType.TEMPLATE &&
-            result.submission.participation.type !== ParticipationType.SOLUTION,
-    );
+export const isStudentParticipation = (participation: Participation) => {
+    return Boolean(participation.type !== ParticipationType.TEMPLATE && participation.type !== ParticipationType.SOLUTION);
 };
 
 /**
  * Returns true if the submission of the result is of type programming, is automatic, and
  * its build has failed.
  * @param result
+ * @param participation
  */
-export const isBuildFailedAndResultIsAutomatic = (result: Result) => {
-    return isBuildFailed(result.submission) && !isManualResult(result);
+export const isBuildFailedAndResultIsAutomatic = (result: Result, participation: Participation) => {
+    const latestSubmission = result?.submission ?? participation?.submissions?.[0];
```

Comment:
Same here `participation?.submissions?.[0]`

This should rather be `participation.getLatestSubmission()`